ifrs17/Import/Importers.ipynb

```diff
@@ -287,6 +287,92 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Import and Execution with Writing Down Logs into the DB"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "using System.IO;",
+                "\nusing Systemorph.Vertex.DataSetReader.Csv;",
+                "\nusing Systemorph.Vertex.Session;",
+                "\nusing Systemorph.Vertex.Import.Builders;",
+                "\nusing Systemorph.Vertex.Import.Mappings;",
+                "\n",
+                "\npublic record StreamWrapper(Stream Stream, bool WillBeReused);",
+                "\n",
+                "\npublic static string ProcessNotification(this object obj) => obj is ActivityMessageNotification amn ? amn.Message : \"\"; ",
+                "\n",
+                "\npublic static async Task<ActivityLog> ExecuteWithStoreActivityAsync(this ImportOptionsBuilder builder, ",
+                "\n                                                                                ISessionVariable session, ",
+                "\n                                                                                IDataSource dataSource)",
+                "\n{",
+                "\n    var log = await builder.ExecuteAsync();",
+                "\n    var options = builder.GetImportOptions();",
+                "\n    string formatFromFile;",
+                "\n    byte[] content;",
+                "\n    StreamWrapper stream = options switch",
+                "\n    {",
+                "\n        FileImportOptions fio => new StreamWrapper(await fio.Storage.ReadAsync(fio.FileName, session.CancellationToken), true),",
```

Comment:
This whole thing looks too low level for a business project.

We need have some abstraction and agree on an API and move this outside.

Reply:
OK, now it has been split into a couple of subroutines and moved to the Utils. For the additional level of abstraction: I define now an abstract class ImportWithKey, that has the properties that are common to all the imports: SerializedContent, creation time length and format. All the actuals imports (file, stream, string, data set) inherit from it and have their own attributes on top of that. All this now have moved to the Utils, while the technical steps are now private methods in the appropriate records. The ExecuteWithStoreActivityAsync method is now rewritten with this high level API.